spec/function.dd

```diff
@@ -2181,15 +2181,49 @@ void main()
 $(H2 $(LNAME2 interpretation, Compile Time Function Execution (CTFE)))
 
     $(P Functions which are both portable and free of global side-effects can be
-    executed at compile time. This is useful when constant folding
-    algorithms need to include recursion and looping. Compile time function
-    execution is subject to the following restrictions:
+    executed at compile time. In certain contexts, such compile time execution
+    is guaranteed. It is called Compile Time Function Execution (CTFE) then.
+    The contexts that trigger CTFE are:)
+
+    $(UL
+    $(LI initialization of a static variable or a manifest constant)
+    $(LI static initializers of struct/class members)
+    $(LI dimension of a static array)
+    $(LI argument for a template value parameter)
+    $(LI $(D_KEYWORD static if))
+    $(LI $(D_KEYWORD static foreach))
+    $(LI $(D_KEYWORD static assert))
+    $(LI $(D_KEYWORD mixin) statement)
     )
 
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+---
+template eval( A... )
+{
+    const typeof(A[0]) eval = A[0];
+}
```

Comment:
Why not simply:
```D
enum eval(Args...) = A[0];
```
?

Reply:
Ah, I see you just moved the example up, but still I think we should show the simplest syntax that works to the readers.